data/card_definitions.json

```diff
@@ -10457,6 +10457,324 @@
+					"amount": 2
+				}
+			]
+		}
+	},
+	{
+		"id": "mole_cavein",
+		"type": "ultra",
+		"display_name": "Cave In",
+		"force_cost": 0,
+		"gauge_cost": 2,
+		"range_min": 1,
+		"range_max": 3,
+		"speed": 1,
+		"power": 4,
+		"armor": 3,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "during_strike",
+				"effect_type": "opponent_cant_move_if_in_range"
+			},
+			{
+				"timing": "after",
+				"effect_type": "draw",
+				"amount": 2
+			}
+		],
+		"boost": {
+			"boost_type": "continuous",
+			"force_cost": 0,
+			"cancel_cost": -1,
+			"display_name": "Undermine",
+			"effects": [
+				{
+					"timing": "start_of_next_turn",
+					"effect_type": "choice",
+					"choice": [
+						{
+							"effect_type": "discard_this",
+							"bonus_effect": {
+								"condition": "opponent_on_buddy_space",
+								"condition_detail": "Burrow",
+								"effect_type": "take_damage",
+								"opponent": true,
+								"nonlethal": true,
+								"amount": 4
+							}
+						},
+						{
+							"effect_type": "move_buddy",
+							"buddy_name": "Burrow",
+							"amount": 1,
+							"amount2": 1
+						}
+					]
+				}
+			]
+		}
+	},
+	{
+		"id": "mole_burrowdig",
+		"type": "special",
+		"display_name": "Burrow Dig",
+		"force_cost": 0,
+		"gauge_cost": 0,
+		"range_min": 1,
+		"range_max": 1,
+		"speed": 7,
+		"power": 3,
+		"armor": 0,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "hit",
+				"effect_type": "draw",
+				"amount": 1,
+				"and": {
+					"effect_type": "advance",
+					"amount": 2
+				}
+			}
+		],
+		"boost": {
+			"boost_type": "immediate",
+			"force_cost": 0,
+			"cancel_cost": -1,
+			"display_name": "Lost Treasure",
+			"effects": [
+				{
+					"timing": "immediate",
+					"effect_type": "choose_discard",
+					"limitation": "special/ultra",
+					"destination": "hand",
+					"amount": 1,
+					"amount_min": 1
+				},
+				{
+					"timing": "immediate",
+					"effect_type": "shuffle_discard_in_place"
+				},
+				{
+					"timing": "immediate",
+					"effect_type": "skip_end_of_turn_draw"
+				}
+			]
+		}
+	},
+	{
+		"id": "mole_headbutt",
+		"type": "special",
+		"display_name": "Headbutt",
+		"force_cost": 1,
+		"gauge_cost": 0,
+		"range_min": 1,
+		"range_max": 2,
+		"speed": 6,
+		"power": 4,
+		"armor": 0,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "hit",
+				"effect_type": "push",
+				"amount": 2,
+				"and": {
+					"effect_type": "choice",
+					"choice": [
+						{ "effect_type": "retreat", "amount": 1 },
+						{
+							"effect_type": "advance",
+							"amount": 2,
+							"and": {
+								"effect_type": "gain_advantage"
+							}
+						}
+					]
+				}
+			}
+		],
+		"boost": {
+			"boost_type": "continuous",
+			"force_cost": 1,
+			"cancel_cost": -1,
+			"display_name": "Come At Me!",
+			"effects": [
+				{
+					"timing": "during_strike",
+					"effect_type": "powerup",
+					"amount": 3
+				},
+				{
+					"timing": "start_of_next_turn",
+					"effect_type": "add_to_gauge_immediately"
```

Comment:
This should have "and draw a card"  too

Reply:
oh huh, this version is better than the S6 version